FEXCore/Source/Interface/IR/Passes/ConstProp.cpp

```diff
@@ -139,23 +137,46 @@ class ConstProp final : public FEXCore::IR::Pass {
 // Constants are pooled per block.
 void ConstProp::HandleConstantPools(IREmitter* IREmit, const IRListView& CurrentIR) {
   const uint32_t SSACount = CurrentIR.GetSSACount();
-  fextl::vector<Ref> Remap(SSACount, NULL);
+
+  // Allocation/initialization deferred until first use, since many multiblocks
+  // don't have constants leftover after all inlining.
+  fextl::vector<Ref> Remap {};
+
+  struct Entry {
+    int64_t Value;
+    Ref R;
+  };
 
   for (auto [BlockNode, BlockIROp] : CurrentIR.GetBlocks()) {
+    fextl::vector<Entry> Pool {};
```

Comment:
This will clear and re-allocate memory in each loop iteration. Instead, declaring the vector outside of the loop and just `.clear()`ing it in each iteration allows to minimize redundant allocations.